### PR TITLE
Move to using verified instructor instead of verified teacher

### PR DESCRIPTION
**Pull request by dmcavoy**
Finishes moving over to using verified instructor instead of verified teacher to check if someone has access to verified content.  Updates `verified_teacher?` to just check if someone has the AUTHORIZED_TEACHER permission.  There are still some places that need to check that someone has that permission specifically. 

## Links

[Eng Plan](https://docs.google.com/document/d/1V61xONU0-mleMEEdHWNxhvZtYIPRCB31mT_hbUXusCQ/edit#)
[Product Spec](https://docs.google.com/document/d/1qZpc90daxQiiqpcQUiUiGV2tQoaYEQUzOA72UZzBlVE/edit#heading=h.f5x8no3lbpxy)

## Testing story

- Updated tests to match updates

### CODE REVIEW

File: dashboard/app/models/script.rb [hunk 1]
```diff
@@ -1577,7 +1577,6 @@ def summarize_header
       isHocScript: hoc?,
       student_detail_progress_view: student_detail_progress_view?,
       age_13_required: logged_out_age_13_required?,
-      is_csf: csf? || csc?, # TODO(dmcavoy): Remove once show_sign_in_callout launched
```
Comment (dmcavoy) [inline dashboard/app/models/script.rb:1565]: Just removing here while I was thinking of it

File: dashboard/app/models/user.rb [hunk 2]
```diff
@@ -1323,13 +1323,7 @@ def teacher?
   end
 
   def verified_teacher?
```
Comment (dmcavoy) [inline dashboard/app/models/user.rb:1330]: We are still using this method but its only to check if someone specifically has this permission

File: dashboard/test/models/script_test.rb [hunk 3]
```diff
@@ -812,7 +812,6 @@ def populate_cache_and_disconnect_db
       isHocScript: false,
       student_detail_progress_view: false,
       age_13_required: false,
-      is_csf: false,
```
Comment (dmcavoy) [inline dashboard/test/models/script_test.rb:815]: This is just updating the tests for removing this variable.

File: dashboard/app/models/user.rb (hunk 2 above)
Comment (jamescodeorg) [reply dashboard/app/models/user.rb:1330]: Maybe add a header comment that indicates how this is different from `verified_instructor?` and that, in most cases, the caller should be calling `verified_instructor?` instead.  Also, might be helpful to add a header comment to `verified_instructor?`.  (I'm just imagining someone seeing these two methods in the future and wondering which they should use.)

Also, mostly curious, what's a scenario where we'd want to check for specifically this permission?

**Review by jamescodeorg — APPROVED** [review]
🎉

**Comment by dmcavoy** [comment]
@jamescodeorg Thanks for the feedback. I'm adding those comments now.

One example of where we check that permission is that we check for the AUTHORIZED_TEACHER permission with PL to see if we should authorize a teacher that attended PL.

**Pull request merged**